kie-server-parent/kie-server-controller/kie-server-controller-mgmt-client/src/main/java/org/kie/server/controller/management/client/KieServerMgmtControllerClient.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.controller.management.client;
+
+import java.io.Closeable;
+
+import org.kie.server.controller.api.service.RuleCapabilitiesService;
+import org.kie.server.controller.api.service.RuntimeManagementService;
+import org.kie.server.controller.api.service.SpecManagementService;
+
+public interface KieServerMgmtControllerClient extends SpecManagementService,
```

Comment:
@cristianonicolai I believe there's already class with the same name [here](https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-server-parent/kie-server-tests/kie-server-integ-tests-common/src/main/java/org/kie/server/integrationtests/controller/client/KieServerMgmtControllerClient.java). If I recall correctly that client was only for integration testing and it wasn't released as  "supported client" for normal users to use.  Does it mean that we're changing that now and from now on we'll be providing "official" kie-server-controller client? Also are there any plans to unify this implementation with that testing client from integration test?

EDIT I probably found the answer below, but will be nice to have this confirmed anyway :) @sutaakar @jakubschwan @jiripetrlik be aware - we're using this client in may places and now it's being moved to different maven artifact.

Reply:
I believe it was moved from the integration tests to this new module.

Reply:
@jhrcek correct, we will provide this new controller management client API. So far in this PR, I only moved things around to reuse what was in use in the integration tests already. Next step is to bring a WebSocket version of this API which will be the recommended protocol.